CMakeLists.txt

```diff
@@ -60,7 +60,6 @@ include(CMakeParseArguments)
 include(IcebergBuildUtils)
 include(IcebergSanitizer)
 include(IcebergThirdpartyToolchain)
-include(GenerateExportHeader)
```

Comment:
Meson does not have this feature. Rather than generate something in the build directory, I created the export header directly in the source and modeled the structure of it after similar visibility.h files in Apache Arrow

Reply:
It also appears that this was being used incorrectly, as shared library builds on Windows are broken on main (they aren't tested in the CMake CI)